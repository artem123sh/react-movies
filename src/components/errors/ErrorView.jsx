import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Footer from '../footer';
import Button from '../basic/Button';
import Logo from '../basic/Logo';
import { BACKGROUND } from '../../theme';

const ErrorViewContainer = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  background: ${BACKGROUND};
  position: relative;
`;

const LogoContainer = styled.div`
  position: absolute;
  top: 5%; 
  left: 5%;
  z-index: 15;
`;

const ErrorViewContent = styled.div`
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const ErrorLabel = styled.span`
  font-size: 3rem;
`;

const Error = styled.span`
  font-size: 12rem;
  font-family: Arial Black;
  text-shadow: -10px 0 red, 10px 0 darkturquoise;
  letter-spacing: 2rem;
`;

const ErrorView = ({ errorLabel, error }) => (
  <ErrorViewContainer>
    <ErrorViewContent>
      <LogoContainer><Logo /></LogoContainer>
      <ErrorLabel>{errorLabel}</ErrorLabel>
      <Error>{error}</Error>
      <Button variant="outlined">Go Back To Home</Button>
    </ErrorViewContent>
    <Footer />
  </ErrorViewContainer>
);

ErrorView.propTypes = {
  errorLabel: PropTypes.string.isRequired,
  error: PropTypes.string.isRequired,
};

export default ErrorView;
