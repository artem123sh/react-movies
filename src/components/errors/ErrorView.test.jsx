import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import ErrorView from './ErrorView';

describe('ErrorView', () => {
  test('renders ErrorView\'s snapshot', () => {
    const { asFragment } = render(<Router><ErrorView errorLabel="label" error="error" /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });
});
