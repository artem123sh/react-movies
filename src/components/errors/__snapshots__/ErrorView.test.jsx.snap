// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView renders ErrorView's snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw cLGzQq"
  >
    <div
      class="sc-AxhUy gZzNyQ"
    >
      <div
        class="sc-AxhCb hceobT"
      >
        <span
          class="sc-AxjAm hgfYjl"
        >
          netflix
        </span>
        <span
          class="sc-AxjAm cIZlLY"
        >
          roulette
        </span>
      </div>
      <span
        class="sc-AxgMl varXG"
      >
        label
      </span>
      <span
        class="sc-AxheI jJLfGq"
      >
        error
      </span>
      <a
        href="/"
      >
        <button
          class="sc-AxirZ gIFDjz"
          type="button"
        >
          Go Back To Home
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
